### ISSUE TITLE
Visualization: Print subquery number of executions, average time, and total time

### ISSUE THREAD

**Issue by mrks** [hyrise/hyrise#1019]
I can't check how it is right now because of #1018, but when I call visualize on a PQP with a subquery, it should show me the number of executions and the avg. and total execution time. Also, it should be clear if the time displayed for the operators is per execution of the subquery or total.

**Comment by mrzzzrm** [comment]
@mrks adding execution time for individual operators of subqueries is an awkward bit of coding, do you insist on it?

**Comment by mrks** [comment]
Could you summarize why it is difficult? My first approach would be to sum it up in `BaseOperatorPerformanceData:walltime` and add a `number_of_executions` there.

**Comment by mrks** [comment]
The number of executions could be part of the edge label ("Subquery, executed once" and "Subquery, executed n times")

**Comment by mrzzzrm** [comment]
The ugly part is having to extract all this info in the ExpressionEvaluator and then writing the averages into the Operators of the PQPSelectExpression.

**Comment by mrzzzrm** [comment]
(The operators that actually execute the subselect never leave the ExpressionEvaluator)

**Comment by mrks** [comment]
I can't follow. The LQPVisualizer knows about them, so they do leave the evaluator, don't they?

**Comment by mrzzzrm** [comment]
I assume you mean the PQPVisualizer (aka SQLQueryPlanVisualizer), and we're talking about PQP visualization.

The PQPSelectExpression holds a "prototype PQP" for the subquery, from which a deep copy is created for each row that it is executed on. The prototype PQP is never executed. 
This is why we would need to do the performance measurement inside the ExpressionEvaluator (which executes the subquery), match all operators in the actually executed PQPs with the operators in the prototype PQP and then assign the average execution times to the operators in the prototype PQP -- in order for them to show up in the PQP visualization.

Talking aside: I'll try to come up with something as I acknowledge that having subquery runtime information in the PQP visualization is useful. It just won't be pretty.

(This is talking about the execution times of the individual operators of subqueries. Execution time of the entire subquery will be easier)

**Comment by mrks** [comment]
Two thoughts about this:
* Once visualized, the subquery execution time should be subtracted from the execution time of the operator that required the subquery. Keep in mind that a subquery might be shared by multiple operators.
* The OperatorPerformanceData might have a link to some "upstream" performance data, which is set during deep_copy, and `AbstractOperator::execute` could add to both.
